src/oneD/StFlow.cpp

```diff
@@ -171,6 +185,17 @@ void StFlow::enableSoret(bool withSoret)
     }
 }
 
+void StFlow::enableAmbi(bool withAmbi)
+{
+    if (!m_do_multicomponent) {
```

Comment:
Is there a reason that this cannot be used with the multicomponent model? If so, then the same check must be applied in `setTransport`.